tools/cmd/wrong-type-detector/main.go

```diff
@@ -0,0 +1,150 @@
+package main
```

Comment:
Can we change the path to this file?

I would change it to `tools/wrong-type-detector/cmd/main.go` as it is the `cmd` of the tool `wrong-type-detector`. I would even remove the `cmd` part...